src/validators/model_fields.rs

```diff
@@ -191,13 +191,18 @@ impl Validator for ModelFieldsValidator {
                             fields_set_vec.push(field.name_py.clone_ref(py));
                             fields_set_count += 1;
                         }
-                        Err(ValError::Omit) => continue,
-                        Err(ValError::LineErrors(line_errors)) => {
-                            for err in line_errors {
-                                errors.push(lookup_path.apply_error_loc(err, self.loc_by_alias, &field.name));
+                        Err(e) => {
+                            state.has_field_error = true;
```

Comment:
Why do we need to track this on the state? Should we adjust the algorithm to just never call `.default_value` if there's any errors?

(We could go further and rearrange things a bit so that all default values only get filled in after the whole input is processed, there is some overlap with #1620 🤔)

Reply:
> Why do we need to track this on the state? Should we adjust the algorithm to just never call .default_value if there's any errors?

It's only relevant for default factories taking the data argument, so if you just have a "static" default it shouldn't matter. And because the logic to call the default value is nested under the current field validator, I can't really "influence" the logic from the model field.

This state approach doesn't look right, but I couldn't find a better way to do so :thinking: 

> (We could go further and rearrange things a bit so that all default values only get filled in after the whole input is processed, there is some overlap with #1620 🤔)

Would this allow using other field values independently from the field order?

Reply:
Ah, I see. I missed the default factories guarantee field order; this makes a lot of sense.

Because of the existence of `PydanticUndefined` as a way to ask for defaults, I think my original proposal actually doesn't make sense. 🤔 

... but I do worry about when this flag is reset. Currently once you set it it's true for the rest of the validation. This seems problematic in unions, for example.

Reply:
Indeed this is one of the reasons it doesn't look right; however, for unions I think it should be alright, as there's a boundary in the validation error.

The error we are handling here is the validation error of the field. If this field is defined like:

```python
class Model(BaseModel):
    a: int | str
```

And a validation error happens on `int`, the field validator will try validating against `str`. At this point, we don't know what happened, and we will only get a val error if validation against `str` failed as well.